analyzer/analyzer.py

```diff
@@ -292,43 +301,61 @@ def format_result_for_visualizer(self, data):
+                        if tmp_group != data_group:
+                            if field_type not in output:
+                                output[field_type] = {}
+                            output[field_type][data_group] = {}
+                            continue
+
+                        orgData = data[node_type][node][field_type][data_group] if data[node_type][node][field_type].has_key(data_group) else data[node_type][node][field_type]
+                        if field_type == "phase":
+                            continue
+                        if field_type not in output:
+                            output[field_type] = OrderedDict()
+                        if data_group not in output[field_type]:
+                            output[field_type][data_group] = OrderedDict()
+                        if node not in output[field_type][data_group]:
+                            output[field_type][data_group][node] = OrderedDict()
+
+                        if "phase" in data[node_type][node].keys() and field_type in phase_name_map.keys():
+                            try:
+                                start = int(data[node_type][node]["phase"][phase_name_map[field_type]]["benchmark_start"])
+                                end = int(data[node_type][node]["phase"][phase_name_map[field_type]]["benchmark_stop"])
+                                benchmark_active_time = end - start
+                                if benchmark_active_time > (rampup + runtime) or end <= 0:
+                                    runtime_end = start + rampup + runtime
+                                else:
+                                    runtime_end = end
+                                runtime_start = start + rampup
+                                runtime_start = runtime_start / monitor_interval
+                                runtime_end = runtime_end / monitor_interval
+
+                                for colume_name, colume_data in orgData.items():
+                                    if isinstance(colume_data, list):
+                                        colume_data = colume_data[runtime_start:runtime_end]
+                                    output[field_type][data_group][node][colume_name] = colume_data
+                            except:
+                                output[field_type][data_group][node] = orgData
+                        else:
+                            output[field_type][data_group][node] = orgData
+
             for key in sorted(output.keys()):
                 output_sort[node_type][key] = copy.deepcopy( output[key] )
-
-        cpu_core_data_dict = OrderedDict()
-        for key,value in output_sort["ceph"]["cpu"].items():
-            for name,data in value.items():
-                cpu_core_data_dict[name] = data
-        output_sort["ceph"]["cpu"] = cpu_core_data_dict
-
+        for data_group in ["summary", "detail"]:
```

Comment:
I think below codes best to be put into process_sar_data function rather than put it here, any comment?